lib/src/robot/client.dart

```diff
@@ -75,7 +80,10 @@ class RobotClient {
   /// Refresh the resources of this robot
   Future<void> refresh() async {
     final ResourceNamesResponse response = await _client.resourceNames(ResourceNamesRequest());
-    if (response.resources == resourceNames) {
+    if (setEquals(response.resources.toSet(), resourceNames.toSet())) {
```

Comment:
(q) why `if(setEquals(response.resources.toSet(),.......` as opposed to `if (response.resources == resourceNames)` ?

Reply:
Mostly because the second way wasn't working. I didn't dive too deep into it but converting both to lists, sorting those lists, making everything lowercase strings, nothing actually make things equal except converting these to sets and using `setEquals`